files/edge-core/scripts/arm_update_activate.sh

```diff
@@ -28,10 +28,12 @@
 mkdir -p "$UPGRADE_DIR"
 
 # copy header to location to be applied by bootloader
-[ -n "$HEADER" ] && cp "$HEADER" "$UPGRADE_HDR"
+[ -f "$HEADER" ] && cp "$HEADER" "$UPGRADE_HDR"
 # copy firmware to location to be applied by bootloader
-[ -n "$FIRMWARE" ] && cp "$FIRMWARE" "$UPGRADE_TGZ"
+[ -f "$FIRMWARE" ] && cp "$FIRMWARE" "$UPGRADE_TGZ"
 # create fake installer.bin until we figure out if it's needed for anything
 # the size originates from struct _arm_uc_installer_details_t in arm_uc_types.h
 # see https://github.com/ARMmbed/mbed-edge/blob/master/lib/mbed-cloud-client/update-client-hub/modules/common/update-client-common/arm_uc_types.h#L74
 dd if=/dev/zero of=$UPGRADE_INS bs=68 count=1
+
+exit 0
```

Comment:
This will mask any error returns from `dd`, is that the intent?

Reply:
I can remove this one so that the return code from dd is returned, but the next one should probably stay.

Reply:
I'll comment on the next one so we can discuss it there. You're right, it's a different scenario.